visualizer/src/service/selectMachine.js

```diff
@@ -148,6 +151,12 @@ const selectMachine = Machine(
       ...selectShortcutActions,
       ...cycleActions,
       ...setActions,
+      save: respond(({ foreground, background }) => ({ type: 'RESTORE', foreground, background })),
+      restore: pure((_, { foreground, background }) => [
+        respond('RESTORED'),
+        send({ type: 'FOREGROUND', foreground }),
+        send({ type: 'BACKGROUND', background }),
+      ]),
```

Comment:
I expected to see a `SAVE` type in here somehwere?

Reply:
`SAVE` events are only sent by the `undoMachine` when an action happens to tell a tracking machine to create a snapshot of itself that it restore in case the action is undone/redone. When sending `SAVE`, the `undoMachine` expects tracked machines to respond with a `RESTORE` event that the `undoMachine` records and sends back to the tracked machine when undoing or redoing the action

Reply:
thank you for the clarification!